lib/axiom/fs/base/open_context.js

```diff
@@ -78,6 +78,32 @@ export default OpenContext;
 OpenContext.prototype = Object.create(Ephemeral.prototype);
 
 /**
+ * Wrap a promise operation on an `OpenContext` instance, ensuring the
+ * context is closed after the promise is settled.
+ *
+ * @template T
+ * @param {!OpenContext} cx
+ * @param {!function(!OpenContext): !Promise<T>} promise
+ * @return {!Promise<T>}
+ */
+OpenContext.scope = function(cx, promise) {
+  return promise(cx).then(
+    function(value) {
+      cx.closeOk();
+      return Promise.resolve(value)
+    },
+    function(error) {
+      if (cx.isEphemeral('Ready')) {
```

Comment:
I'm a bit confused what's going on here.  A comment might be in order (as to why cx.closeOk() is getting called on a promise rejection)